config/src/main/java/com/epam/aidial/core/config/databind/JsonArrayToSchemaMapDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.epam.aidial.core.config.databind;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class JsonArrayToSchemaMapDeserializer extends JsonDeserializer<Map<String, String>> {
+
+    @Override
+    public Map<String, String> deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        TreeNode tree = jsonParser.readValueAsTree();
+        if (!tree.isArray()) {
+            throw InvalidFormatException.from(jsonParser, "Expected a JSON array of schemas", tree.toString(), Map.class);
+        }
+        Map<String, String> result = new HashMap<>();
+        for (int i = 0; i < tree.size(); i++) {
+            TreeNode value = tree.get(i);
+            if (!value.isObject()) {
```

Comment:
check that is not null, in case of null element?

Reply:
What does mean null in this array?

Reply:
Make sense. Fixed.